src/process/process.rs

```diff
@@ -76,15 +75,29 @@ struct LoadInfo {
     bss_end_offset: usize    /* Offset to end of BSS */
 }
 
+struct MemorySlice<T> {
+    data: T,
+    len: usize,
+}
+
+impl<T> MemorySlice<*mut T> {
+    fn from_mut_slice(s: &'static mut [T]) -> MemorySlice<*mut T> {
+        MemorySlice {
+            data: s.as_mut_ptr(),
+            len: s.len(),
+        }
+    }
+}
+
 pub struct Process<'a> {
     /// The process's memory.
-    memory: Slice<u8>,
+    memory: MemorySlice<*mut u8>,
```

Comment:
This is a perfectly reasonable solution, but maybe we can just replace this with a `&'static mut [u8]` (which is basically the same thing)? I think the only reason not to would be if it would significantly complicate the rest of the code.